frame/society/src/lib.rs

```diff
@@ -1218,29 +1272,21 @@ impl<T: Config<I>, I: Instance> Module<T, I> {
 				// If the element is not at the end of the list, insert the new element
 				// in the spot.
 				if let Some((p, _)) = different_bid {
-					bids.insert(p, Bid {
-						value,
-						who: who.clone(),
-						kind: bid_kind,
-					});
+					insert_pos = Some(p);
+					bids.try_insert(p, bid_to_insert.clone())
 				// If the element is at the end of the list, push the element on the end.
 				} else {
-					bids.push(Bid {
-						value,
-						who: who.clone(),
-						kind: bid_kind,
-					});
+					bids.try_push(bid_to_insert.clone())
 				}
 			},
-			Err(pos) => bids.insert(pos, Bid {
-				value,
-				who: who.clone(),
-				kind: bid_kind,
-			}),
-		}
+			Err(pos) => {
+				insert_pos = Some(pos);
+				bids.try_insert(pos, bid_to_insert.clone())
+			}
+		};
 		// Keep it reasonably small.
-		if bids.len() > MAX_BID_COUNT {
-			let Bid { who: popped, kind, .. } = bids.pop().expect("b.len() > 1000; qed");
+		if res.is_err() {
```

Comment:
This logic change is also not right.

The way the logic was before ensured the smallest value is removed since we added the value in order first, and then popped the last value.

Now, we potentially pop first, and then add the item after, which means the smallest item could pop the second smallest item.

So not a good change here